sdk/core/azure-core-amqp/src/models/amqp_value.cpp

```diff
@@ -0,0 +1,865 @@
+      case AmqpValueType::Ubyte:
+        return static_cast<uint8_t>(*this) < static_cast<uint8_t>(that);
+      case AmqpValueType::Ushort:
+        return static_cast<uint16_t>(*this) < static_cast<uint16_t>(that);
+      case AmqpValueType::Uint:
+        return static_cast<uint32_t>(*this) < static_cast<uint32_t>(that);
+      case AmqpValueType::Ulong:
+        return static_cast<uint64_t>(*this) < static_cast<uint64_t>(that);
+      case AmqpValueType::Byte:
+        return static_cast<std::int8_t>(*this) < static_cast<std::int8_t>(that);
+      case AmqpValueType::Short:
+        return static_cast<int16_t>(*this) < static_cast<int16_t>(that);
+      case AmqpValueType::Int:
+        return static_cast<int32_t>(*this) < static_cast<int32_t>(that);
+      case AmqpValueType::Long:
+        return static_cast<int64_t>(*this) < static_cast<int64_t>(that);
+      case AmqpValueType::Float:
+        return static_cast<float>(*this) < static_cast<float>(that);
+      case AmqpValueType::Double:
+        return static_cast<double>(*this) < static_cast<double>(that);
+      case AmqpValueType::Char:
+        return static_cast<char32_t>(*this) < static_cast<char32_t>(that);
+      case AmqpValueType::String:
+        return static_cast<std::string>(*this) < static_cast<std::string>(that);
+      case AmqpValueType::Symbol:
+        return AsSymbol() < that.AsSymbol();
+
+      case AmqpValueType::Map:
+        return AsMap() < that.AsMap();
+      case AmqpValueType::Array:
+        return AsArray() < that.AsArray();
+      case AmqpValueType::Timestamp:
+        return AsTimestamp() < that.AsTimestamp();
+      case AmqpValueType::Uuid:
+        return static_cast<Azure::Core::Uuid>(*this).AsArray()
+            < static_cast<Azure::Core::Uuid>(that).AsArray();
+      case AmqpValueType::Binary:
+        return AsBinary() < that.AsBinary();
+      case AmqpValueType::List:
+        return AsList() < that.AsList();
+      case AmqpValueType::Described:
+        return AsDescribed() < that.AsDescribed();
+      case AmqpValueType::Composite:
+        return AsComposite() < that.AsComposite();
+      default:
+        throw std::logic_error("Unknown Amqp Value type in operator<");
+    }
+  }
+
+  AmqpMap AmqpValue::AsMap() const { return AmqpMap(m_value.get()); }
+
+  AmqpArray AmqpValue::AsArray() const { return AmqpArray(m_value.get()); }
+
+  AmqpSymbol AmqpValue::AsSymbol() const { return AmqpSymbol(m_value.get()); }
+
+  AmqpComposite AmqpValue::AsComposite() const { return AmqpComposite(m_value.get()); }
+  AmqpList AmqpValue::AsList() const { return AmqpList(m_value.get()); }
+  AmqpBinaryData AmqpValue::AsBinary() const { return AmqpBinaryData(m_value.get()); }
+  AmqpDescribed AmqpValue::AsDescribed() const { return AmqpDescribed(m_value.get()); }
+  AmqpTimestamp AmqpValue::AsTimestamp() const { return AmqpTimestamp(m_value.get()); }
+
+  AmqpValueType AmqpValue::GetType() const
+  {
+    switch (amqpvalue_get_type(m_value.get()))
```

Comment:
Turn into map?

Reply:
I hadn't considered it, but it is much nicer.